### PR TITLE
feat: show more detailed provisioner version info

### LINKED ISSUE #14438: ui(provisioners): proper display of version info
Display the version information of each provisioner/group in the UI.

- Each provisioner should either show an "Up to date" or "Out of date" message, which the user can hover over for more details about the version.

- Groups in which all provisioners are running the same version can show the version number at a higher level, and groups with multiple versions should just say "Multiple versions"

### PR DESCRIPTION
**Pull request by aslilac**
Closes #14438

Screenshot 2024-09-06 at 12 37 01 PM

### CODE REVIEW

File: site/src/modules/provisioners/ProvisionerGroup.tsx [hunk 1]
```diff
@@ -211,3 +255,19 @@ export const ProvisionerGroup: FC<ProvisionerGroupProps> = ({
...
 	);
 };
+
+const styles = {
+	title: (theme) => ({
+		marginTop: 0,
+		marginBottom: 0,
+		color: theme.palette.text.primary,
+		fontSize: 14,
+		lineHeight: "150%",
```
Comment (Parkreiner) [inline site/src/modules/provisioners/ProvisionerGroup.tsx:265]: Could probably be swapped for a value of `1.5`

File: site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.stories.tsx [hunk 2]
```diff
@@ -26,6 +26,19 @@ export const Provisioners: Story = {
...
+						MockProvisioner,
+						{
+							...MockProvisioner2,
+							version: "2.0.0",
+							api_version: "1.0",
+							warnings: [{ code: "EUNKNOWN", message: "私は時代遅れです" }],
+						},
+					],
+				],
+				["じぇいダレン", [MockProvisioner, MockProvisioner2]],
```
Comment (Parkreiner) [inline site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.stories.tsx:41]: Not sure what word this is supposed to be. If this is Jaayden, it should be `ジェイデン`

File: site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.stories.tsx [hunk 3]
```diff
@@ -26,6 +26,19 @@ export const Provisioners: Story = {
...
 			keys: new Map([
+				[
+					"ベン",
+					[
+						MockProvisioner,
+						{
+							...MockProvisioner2,
+							version: "2.0.0",
+							api_version: "1.0",
+							warnings: [{ code: "EUNKNOWN", message: "私は時代遅れです" }],
```
Comment (Parkreiner) [inline site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.stories.tsx:37]: あ、そうそうそう

**Review by buenos-nachos — COMMENTED** [review]
Looks good!
Just messaged you on Slack to see how you want me to handle approvals, but this should be good to approve

**Review by buenos-nachos — APPROVED** [review]
Approved

File: site/src/modules/dashboard/Navbar/UserDropdown/UserDropdown.stories.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ const Example: Story = {
 		await step("click to open", async () => {
 			await userEvent.click(canvas.getByRole("button"));
 			await waitFor(() =>
-				expect(screen.getByText(/v99\.999\.9999/i)).toBeInTheDocument(),
+				expect(screen.getByText(/v2\.99\.99/i)).toBeInTheDocument(),
```
[the remark this answers is not in the document]
Comment (aslilac) [reply site/src/modules/dashboard/Navbar/UserDropdown/UserDropdown.stories.tsx:34]: I like that idea!

File: site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.stories.tsx (hunk 2 above)
Comment (aslilac) [reply site/src/pages/ManagementSettingsPage/OrganizationProvisionersPageView.stories.tsx:41]: my japanese input was being weird when I typed this and I didn't notice how much it butchered the end result 💀 it was indeed supposed to be ジェイデン

**Pull request merged**